apps/obs-web/src/views/AttributeSearchComp.vue

```diff
@@ -0,0 +1,239 @@
+<template>
+  <div>
+      <div class="row q-gutter-xs">
+        <q-select
+            class="col"
+            label="Attribute"
+            outlined
+            dense
+            options-dense
+            :options="searchTypes"
+            v-model="selectedSearchType"
+            clearable
+        ></q-select>
+        <q-select
+            class="col"
+            label="Is"
+            dense
+            outlined
+            options-dense
+            :options="['equals', 'greater than', 'less than']"
+            v-model="evaluator"
+        ></q-select>
+        <q-select
+            v-if="['departureDate', 'returnDate'].includes(selectedSearchType)"
+            class="col"
+            label="Date Selection"
+            outlined
+            dense
+            options-dense
+            use-input
+            :options="searchOptions"
+            v-model="selectedSearchOption"
+            @filter="selectionFilterFn"
+            clearable
+            :display-value="getOptionLabel(selectedSearchOption)"
+            :option-label="opt => getOptionLabel(opt)"
+            :option-value="opt => opt"
+        >
+            <template v-slot:append>
+                <q-icon name="event" class="cursor-pointer">
+                <q-popup-proxy ref="qDateProxy" transition-show="scale" transition-hide="scale">
+                    <q-date v-model="dateFormatter">
+                    <div class="row items-center justify-end">
+                        <q-btn v-close-popup label="Close" color="primary" flat />
+                    </div>
+                    </q-date>
+                </q-popup-proxy>
+                </q-icon>
+            </template>
+        </q-select>
+        <q-select
```

Comment:
@melinashak-noaa I finally have this figured out... as I understand it, when couch indexes a view, all of the emits for a doc are run concurrently, so if one fails, they all fail.  One of my emits was for buyers, and while I handled no buyers array in the doc, and an empty buyers array, I hadn't handled the condition on Mackenzie McCoy's docs - a buyers array with a null in it.  I've updated my view code, so you should now see all of those trips.

A couple things I came across:
1. there are some bad attributes in 916975c39b316dfdbece7e2c417d2e43, with commas instead of periods, like "tripStatus,description", and "departurePort,name" - I'm sure this is just a byproduct of a coding typo, but wanted to make you aware in case the typo still exists or the doc is causing issues.

2. There are 5 documents returned by using the Mackenzie McCoy observer id as a key to the wcgop_trips_by_observerid view.  When I look at the observer objects within those trips, there are 3 different firstname lastname combos: "Mackenzie McCoy", "Mackenzier McCoy", and "melinae Shak".  Is this a product of the trip storing the values of the observer doc at the time the trip was created, and the observer doc was updated a few times?  Or is it somehow possible to change the firstname and lastname attributes associated with a trip without associating a different observer?  Just wanted to raise this as a potential complication to searching for an observer by name vs finding trips by observer id.  Would it be useful for an observer to have a "friendly" ID that's easily searchable + typeable?

The search component isn't loading any docs, so should be pretty fast.  The slowness comes in when the table tries to load 10+ thousand docs.  I believe this is happening in DebrieferTrips.vue line 404 - getTrips()  I know the debriefer folks didn't like pagination, but maybe we should paginate results at a high number of rows, maybe 250 or 500?

Reply:
Oh got it, I see, yes I'll take a look at the new view then. 

1. Oh okay, thanks for pointing that out. Yes that was probably caused by a time I was messing with the a json formatting library and had it accidentally use commas as the delimiter instead of periods. It's been fixed since then so should no longer be happening. Yes I'll probably look into fixing any documents with that odd format. 

2. So that was caused when I was testing out the edit feature on the debriefer tool and updating the observer name. Since then we've updated the tool so the observer name is no longer editable so we should not run into these issues of multiple names in the future. 

Oh yes good point hadn't thought of that, yes it would probably be good to do some lazy loading then so they don't have to wait for all the entries to load before viewing them.